sansio_lsp_client/structs.py

```diff
@@ -51,11 +54,15 @@ class VersionedTextDocumentIdentifier(TextDocumentIdentifier):
     version: t.Optional[int]
 
 
+# Sorting tip:  sorted(positions, key=(lambda p: p.as_tuple()))
```

Comment:
Should we add some comparison operators here so that this is redundant?

Reply:
There used to be comparison operators (actually just `__lt__`), but I switched to this because it seemed a lot simpler. Imagine seeing this in the middle of longer code:
- `sorted(positions)` is like "Can you really do this? Does it work? Is this a typo?"
-  `sorted(positions, key=...)` is like "We compare them as tuples, so it works"

Before `.as_tuple()`, there was also a convenience method that returned `foo.range.start.as_tuple()` (I think), and that was replaced by `.as_tuple()` too. Overall `.as_tuple()` feels like a nice and minimal approach to me.

Reply:
Understandable.